.changelog/26025.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+client: fix batch job drain behavior
```

Comment:
This should be a user-facing changelog item. Something like (this is simultaneously a bit wordy and a bit incomplete though):

```suggestion
drain: Fixed a bug where batch allocations that don't complete by the drain deadline would not be rescheduled later
```